Core/EventHandlers.lua

```diff
@@ -1361,6 +1353,27 @@ function R:OnEvent(event, ...)
 			end
 		end
 
+		-- Handle opening lots of various chests for Gilded Wader (pet).
+		if
+			Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and
+				(Rarity.lastNode == L["Gift of Thenios"] or Rarity.lastNode == L["Hidden Hoard"]
```

Comment:
A suggestion to make it more readable: Use a lookup table, similar to a switch statement?

Example:

```
local nodes = {
  [L["Gift of Thenios"]] = true,
  [L["Hidden Hoard"]] = true,
}
```

and then

```
local isRelevantNode = nodes[Rarity.lastNode] -- maybe check if it's not nil before using it as an index?
if isRelevantNode then
  -- do stuff here
end
```

It can probably made more readable with functions or better naming, but essentially the switch cases can be outsourced to a table and that can be moved anywhere to clean up the code later.

Reply:
Updated to this in 7dbb520f95ba74682c4acd0300a0dc775db318ce